drivers/dma/Kconfig.mcux_edma

```diff
@@ -6,13 +6,28 @@
 config DMA_MCUX_EDMA
 	bool "MCUX DMA driver"
 	default y
-	depends on DT_HAS_NXP_MCUX_EDMA_ENABLED
+	depends on DT_HAS_NXP_MCUX_EDMA_ENABLED \
+		|| DT_HAS_NXP_MCUX_EDMA_V3_ENABLED
 	imply NOCACHE_MEMORY if HAS_MCUX_CACHE
 	help
 	  DMA driver for MCUX series SoCs.
 
 if DMA_MCUX_EDMA
 
+config DMA_MCUX_EDMA_LEGACY
+	bool
+	default y
+	depends on DT_HAS_NXP_MCUX_EDMA_ENABLED
+	help
+	  Enable legacy EDMA for MCUX series SoCs.
```

Comment:
I'm curious about the reason to name this "legacy" and not something like EDMA V2 or V1. If later there's a new EDMA version V4 let's say, would that turn V3 into legacy? I'd expect legacy is a piece of sw/hw that was superseded but somehow we need to keep maintaining. I'm not familiar with the platforms supported by MCUX HAL, but is expected that the new platforms will not use anymore this "legacy" EDMA version?

Reply:
What i understand is that the current EDMA is second version of the DMA IP, the EDMA in S32K344 is third version of the DMA IP. Not sure what is better, should either i kept the current version in upstream as it is (edma) and use dmav3 for the S32K344 or something like edmav2 and edmav3

Reply:
if the existing is V2 and the one you're adding is V3, I'd name it V2 and V3 respectively. But I don't have a strong opinion about it. I'll let other comments if somebody have a strong request to change it.

Reply:
@mmahadevan108 @dleach02 do you have any preference on this? otherwise if you are okay with proposed naming here we can close this comment

Reply:
@manuargue , I agree.

Reply:
okay, i've updated